smack-java7/src/main/java/org/jivesoftware/smack/java7/XmppHostnameVerifier.java

```diff
@@ -133,7 +136,20 @@ private static void matchDns(String name, X509Certificate cert) throws Certifica
                 throw new CertificateException(sb.toString());
             }
         }
-        // TODO SubjectX500Name
+        try {
+            LdapName dn = new LdapName(cert.getSubjectX500Principal().getName());
+            for (Rdn rdn : dn.getRdns()) {
+                if (rdn.getType().equalsIgnoreCase("CN")) {
+                    if (matchesPerRfc2818(name, rdn.getValue().toString())) {
+                        return;
+                    }
+                    break;
+                }
+            }
+        } catch (Exception e) {
```

Comment:
LdapName ctor may throw InvalidNameException. Should the catch be more specific or the exception propagated upwards?